AbletonMCP_Remote_Script/handlers/devices.py

```
@@ -0,0 +1,1381 @@
+    except Exception as e:
+        if ctrl:
+            ctrl.log_message("Error getting macro values: " + str(e))
+        raise
+
+
+def set_macro_value(song, track_index, device_index, macro_index, value, track_type="track", ctrl=None):
+    """Set the value of a specific macro control on a rack device."""
+    try:
+        _track, device = _resolve_device(song, track_index, device_index, track_type)
+        if not hasattr(device, "macros_mapped"):
+            raise Exception("Device is not a rack (no macros)")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)